html/captive-portal/lib/captiveportal/PacketFence/DynamicRouting/Module/Authentication/OAuth/Instagram.pm

```diff
@@ -0,0 +1,178 @@
+package captiveportal::PacketFence::DynamicRouting::Module::Authentication::OAuth::Instagram;
+
+=head1 NAME
+
+captiveportal::DynamicRouting::Module::Authentication::OAuth::Instagram
+
+=head1 DESCRIPTION
+
+Instagram OAuth module
+
+=cut
+
+use WWW::Curl::Easy;
+use Moose;
+use pf::log;
+extends 'captiveportal::DynamicRouting::Module::Authentication::OAuth';
+
+has '+source' => (isa => 'pf::Authentication::Source::InstagramSource');
+
+=head2 get_token
+
+Get the OAuth2 token
+
+=cut
+
+sub get_token {
+    my ($self) = @_;
+
+    my $curl = WWW::Curl::Easy->new;
+    my $code = $self->app->request->parameters->{code};
+    my $info = $self->get_client;
+
+    my $source = $self->source;
+
+    my $response_body = '';
+    my $formdata = WWW::Curl::Form->new;
+    
+    $formdata->formadd("client_id", "$info->{NOP_id}");
+    $formdata->formadd("client_secret", "$info->{NOP_secret}");
+    $formdata->formadd("redirect_uri", "$info->{NOPW_redirect}");
+    $formdata->formadd("grant_type", "authorization_code");
+    $formdata->formadd("code", "$code");
+
+    $curl->setopt(CURLOPT_HTTPPOST, $formdata);
+    $curl->setopt(CURLOPT_WRITEDATA, \$response_body);
+    $curl->setopt(CURLOPT_DNS_USE_GLOBAL_CACHE, 0);
+    $curl->setopt(CURLOPT_NOSIGNAL, 1);
+    $curl->setopt(CURLOPT_HTTPHEADER(), ['Content-Type: multipart/form-data']);
+    $curl->setopt(CURLOPT_URL, $info->{NOP_access_token_url});
+
+    my $curl_return_code = $curl->perform;
+
+    my $response_code = $curl->getinfo(CURLINFO_HTTP_CODE);
```

Comment:
Check that $curl_return_code is good and $response_code.

If the request fails, now it will just not log anything and it will fail after